presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -815,4 +815,76 @@ public static Slice toUtf8(@SqlType("varchar(x)") Slice slice)
     {
         return slice;
     }
+
+    @Description("suffix starting at given index")
+    @ScalarFunction("substr")
+    @SqlType(StandardTypes.VARBINARY)
+    public static Slice varbinarySubstr(@SqlType(StandardTypes.VARBINARY) Slice varbinary, @SqlType(StandardTypes.BIGINT) long start)
```

Comment:
- Let's call the variable `slice`; same for the one below.
- Why don't we just call the function `substr` and remove the `ScalarFunction` annotation above.
- Can we re-use the function instead of writing two pieces of code that look almost the same. We can delegate this function call to the one below with `length = slice.length()`.

Reply:
Agreed with the first two bullet points. 

> Can we re-use the function instead of writing two pieces of code that look almost the same. We can delegate this function call to the one below with length = slice.length().

Logically, the following two functions are not equivalent, though the results may be the same.
`substr(slice, start); //return a suffix from start index`
`substr(slice, start, slice.length()); // return slice.length() bytes from start index`

Reply:
`Why don't we just call the function substr and remove the ScalarFunction annotation above.`: this is not possible because the functions will have the exact same Java function signature.

Reply:
You both make sense. The difference is if the substr for varbinary lives in the same file as  substr for varchar. I see James' point of moving substr for varbinary to VarbinaryFunctions.